qt-models/divetripmodel.cpp

```diff
@@ -693,7 +693,7 @@ dive *DiveTripModelTree::diveOrNull(const QModelIndex &index) const
 
 // Send data changed signals blockwise. Which entries are changed is collected
 // in the first parameter "changed".
-void DiveTripModelBase::sendShownChangedSignals(const std::vector<char> &changed, int parentIndex)
+void DiveTripModelBase::sendShownChangedSignals(const std::vector<char> &changed, quintptr parentIndex)
```

Comment:
I've never noticed us using `quintptr` - what's the advantage over `int *` ?

Reply:
quintptr is what Qt wants in its API:
https://doc.qt.io/qt-5/qabstractitemmodel.html#createIndex-1

This should have read quintptr from the start - I will rebase accordingly.

Reply:
Note that, in contrast to what the name might imply, `quintptr` is an integer type, *not* a pointer. That's why we are using this here over the `void *` version. It's Qt's version of the C99 `uintptr_t` type.

Reply:
mind boggles